src/components/Layout/index.tsx

```diff
@@ -0,0 +1,14 @@
+import type { ReactNode, FC } from "react";
+import { Header } from "./Header";
+
+type Props = {
+  children: NonNullable<ReactNode>;
```

Comment:
I completely understand what you did here. Just for the sake of simplicity, it is fine to let `null` and `undefined` here. 

Be aware of the difference between

`children?: ReactNode`

and
                                      (added the undefined here explicitly, I know it is part of it)
`children: ReactNode | undefined`


`?` means that the prop is optional, **it can be missing**

having `undefined` as a type means it has to be explicitly marked as undefined

![CleanShot 2023-04-04 at 17 56 29](https://user-images.githubusercontent.com/12488648/229849219-d4eba369-e97b-46e3-b473-8b68f461b145.png)

Writing {undefined} in JSX is something you would notice in code review. That is why we are probably fine with it.